src/TTimer.cpp

```diff
@@ -86,15 +85,18 @@ void TTimer::setName(const QString& name)
         mpHost->getTimerUnit()->mLookupTable.remove(mName, this);
     }
     mName = name;
+    // Merely for information if needed later:
+    mpQTimer->setObjectName(QStringLiteral("timer(Host:%1)(TTimerId:%2)").arg(mpHost->getName(), name));
     mpHost->getTimerUnit()->mLookupTable.insertMulti(name, this);
 }
 
 void TTimer::setTime(QTime time)
 {
     QMutexLocker locker(&mLock);
+    // Stop the timer before doing anything else:
+    mpQTimer->stop();
     mTime = time;
-    mpTimer->setInterval(mTime.msec() + (1000 * mTime.second()) + (1000 * 60 * mTime.minute()) + (1000 * 60 * 60 * mTime.hour()));
-    mpTimer->stop();
+    mpQTimer->setInterval(time.msecsSinceStartOfDay());
```

Comment:
why was this changed? it seems to do a different thing, no?

Reply:
Nope - it is a simpler/more elegant command that does the same thing (in one call) - it returns the time since `0:00:00.000` in milli-seconds and what it returns which is what the previous calculation did... :nerd_face:

Reply:
I'm confused because https://doc.qt.io/qt-5/qtime.html#msec does not mention time of the day

Reply:
But https://doc.qt.io/qt-5/qtime.html#msecsSinceStartOfDay does:

> ... Returns the number of msecs since the start of the day, i.e. since 00:00:00.

... and since we are using the class to store an *interval* it works out to be exactly what we want.

Reply:
Ah! My bad, sorry for being difficult.